bluemira/equilibria/fem_fixed_boundary/utilities.py

```diff
@@ -59,30 +67,192 @@ class ScalarSubFunc(dolfin.UserExpression):
 
     def value_shape(self):
         """
         Value_shape function (necessary for a UserExpression)
         https://fenicsproject.discourse.group/t/problems-interpolating-a-userexpression-and-plotting-it/1303
         """
         return ()
+
+
+def plot_scalar_field(
+    x, y, data, levels=20, axis=None, contour=True, tofill=True, **kwargs
+):
+    """
+    Plot a scalar field
+
+    Parameters
+    ----------
+    x: Iterable
+        x coordinate of the points in which
+    """
+    cntr = None
+    cntrf = None
+
+    if axis is None:
+        fig = plt.figure()
+        axis = fig.add_subplot()
+
+    if not kwargs:
+        kwargs = {"linewidths": 2, "colors": "k"}
+
+    # # ----------
+    # # Tricontour
+    # # ----------
+    # # Directly supply the unordered, irregularly spaced coordinates
+    # # to tricontour.
+    # opts = {'linewidths': 0.5, 'colors':'k'}
+    if contour:
+        cntr = axis.tricontour(x, y, data, levels=levels, **kwargs)
+
+    if tofill:
+        cntrf = axis.tricontourf(x, y, data, levels=levels, cmap="RdBu_r")
+        plt.gcf().colorbar(cntrf, ax=axis)
+
+    plt.gca().set_aspect("equal")
+
+    return axis, cntr, cntrf
+
+
+def plot_profile(x, prof, var_name, var_unit):
+    """
+    Plot profile
+    """
+    fig, ax = plt.subplots()
+    ax.plot(x, prof)
+    ax.set(xlabel="x (-)", ylabel=var_name + " (" + var_unit + ")")
+    ax.grid()
+    plt.show()
+
+
+def get_tricontours(x, z, array, value):
+    """
+    Get the contours of a value in a triangular set of points.
+
+    Parameters
+    ----------
+    x: np.array(n, m)
+        The x value array
+    z: np.array(n, m)
+        The z value array
+    array: np.array(n, m)
+        The value array
+    value: float
+        The value of the desired contour in the array
+
+    Returns
+    -------
+    value_loop: np.array(ni, mi)
+        The points of the value contour in the array
+    """
+
+    tri = Triangulation(x, z)
+    tcg = TriContourGenerator(tri.get_cpp_triangulation(), array)
+    return tcg.create_contour(value)[0][0]
+
+
+def calculate_plasma_shape_params(points, psi, levels):
+    """
+    Calculate the plasma parameters (Rgeo, kappa, delta) for a given magnetic
+    isoflux.
+
+    Parameters
+    ----------
+    points: Iterable
+        2D points on which psi has been calculated
+    psi: Iterable
+        scalar value of the plasma magnetic poloidal flux at points
+    levels: Iterable
+        values that identify the isoflux curve at which the plasma parameters are
+        calculated
+
+    Returns
+    -------
+    r_geo: Iterable
+        array of averaged radial coordinate of the isoflux curves
+    kappa:
+        array of the kappa value of the isoflux curves
+    delta:
+        array of the delta value of the isoflux curves
+    """
+    r_geo = np.zeros(len(levels))
+    kappa = np.zeros(len(levels))
+    delta = np.zeros(len(levels))
+
+    tri = Triangulation(points[:, 0], points[:, 1])
```

Comment:
I factored out the direct calls to `matplotlib.pyplot.Axis` this is a similar trick we use in `equilibria/find.py::get_contours`

I only did this because I couldn't work out why it was still plotting with `plot=False` :D